tests/python/gaia-ui-tests/gaiatest/tests/functional/messages/test_delete_thread.py

```diff
@@ -0,0 +1,58 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette import SkipTest
+from marionette_driver import Wait
+from gaiatest import GaiaTestCase
+from gaiatest.apps.messages.app import Messages
+from gaiatest.apps.system.app import System
+from gaiatest.utils.plivo.plivo_util import PlivoUtil
+
+
+class TestSms(GaiaTestCase):
+
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+
+        try:
+            self.testvars['plivo']
+        except KeyError:
+            raise SkipTest('Plivo account details not present in test variables')
+
+        plivo = PlivoUtil(
+            self.testvars['plivo']['auth_id'],
+            self.testvars['plivo']['auth_token'],
+            self.testvars['plivo']['phone_number']
+        )
+
+        plivo.send_sms(to_number=self.environment.phone_numbers[0].replace('+', ''),
+                       message=self._generate_text())
+
+        system = System(self.marionette)
+        system.wait_for_notification_toaster_displayed(timeout=300,
+                                                       message="Notification did not appear. SMS database dump: %s " % self.data_layer.get_all_sms())
```

Comment:
Wasn't the problem with this not that the message is compiled at the time that the wait_for_notification_toaster_displayed function is called, so the SMS database dump is from before the timeout of 300s?